File: examples/sdk/typescript/chat_plan.mjs
#!/usr/bin/env node
import WebSocket from 'ws';
import { SoulBrowserClient } from '../../sdk/typescript/dist/index.js';

const baseUrl = process.env.SOULBROWSER_API_BASE ?? 'http://127.0.0.1:8801';
const prompt = process.argv.slice(2).join(' ') || '打开 https://example.com 并截图';

async function run() {
  const client = new SoulBrowserClient({ baseUrl, WebSocketClass: WebSocket });
  const chat = await client.chat({
    prompt,
    execute: true,
    capture_context: true,
  });

  console.log('[chat] success:', chat.success);
  const plan = chat.plan?.plan;
  if (plan) {
    console.log('[plan] steps:', plan.steps?.length ?? 0);
  }

  const taskId = plan?.task_id;
  if (taskId) {
    console.log('[stream] watching task', taskId);
    const socket = client.openTaskStream(taskId);
    socket.on('message', (event) => {
      try {
        const payload = JSON.parse(event.toString());
        if (payload.event === 'overlay') {
          console.log('[overlay]', payload.overlay.source, payload.overlay.data?.bbox);
        }
      } catch (err) {
        console.error('[stream] failed to parse event', err);
      }
    });
    socket.on('close', () => process.exit(0));
  } else {
    console.log('No task id returned; nothing to stream.');
  }
}

run().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: examples/sdk/python/chat_plan.py
#!/usr/bin/env python3
import argparse
import json
import os
import sys
import time

from soulbrowser_sdk import SoulBrowserClient


def main() -> None:
    parser = argparse.ArgumentParser(description="SoulBrowser Python SDK example")
    parser.add_argument('--base-url', default=os.environ.get('SOULBROWSER_API_BASE', 'http://127.0.0.1:8801'))
    parser.add_argument('--prompt', default='打开 https://example.com 并截图')
    args = parser.parse_args()

    client = SoulBrowserClient(base_url=args.base_url)
    try:
        chat = client.chat({
            'prompt': args.prompt,
            'execute': True,
            'capture_context': True,
        })
        plan = chat.get('plan', {}).get('plan', {})
        print('[chat] success:', chat.get('success'))
        print('[plan] steps:', len(plan.get('steps', [])))
        task_id = plan.get('task_id')
        if not task_id:
            print('No task id returned; exiting')
            return
        print('[stream] watching task', task_id)
        ws = client.open_task_stream(task_id)
        try:
            for event in client.iter_task_stream(ws):
                if event.get('event') == 'overlay':
                    overlay = event.get('overlay', {})
                    print('[overlay]', overlay.get('source'), overlay.get('data'))
                elif event.get('event') == 'status' and event.get('status', {}).get('status') in {'success', 'failed'}:
                    break
        finally:
            ws.close()
    finally:
        client.close()


if __name__ == '__main__':
    main()
